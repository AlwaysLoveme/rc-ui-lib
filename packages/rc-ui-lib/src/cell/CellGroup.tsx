import React from 'react';
import classnames from 'classnames';
import { CellGroupProps } from './PropsType';
import { createNamespace } from '../utils';
import { BORDER_TOP_BOTTOM } from '../utils/constant';

const [bem] = createNamespace('cell-group');

const CellGroup: React.FC<CellGroupProps> = (props) => {
  const { title, border, inset } = props;

  const renderGroup = () => (
    <div className={classnames(bem({ inset }), { [BORDER_TOP_BOTTOM]: !inset && border })}>
      {props.children}
    </div>
  );

  const renderTitle = () => <div className={classnames(bem('title'))}>{title}</div>;

  if (title)
    return (
      <>
        {renderTitle()}
        {renderGroup()}
      </>
    );

  return renderGroup();
};

CellGroup.defaultProps = {
  border: true,
};

export default CellGroup;
