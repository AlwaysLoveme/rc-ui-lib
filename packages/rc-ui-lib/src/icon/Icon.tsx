import React, { useMemo } from 'react';
import classnames from 'classnames';
import { addUnit } from '../utils';
import { IconProps } from './PropsType';
import Badge from '../badge';

function isImage(name?: string): boolean {
  return name ? name.indexOf('/') !== -1 : false;
}

const Icon: React.FC<IconProps> = (props) => {
  const { tag, name, className, onClick, onTouchStart } = props;

  const imageIcon = isImage(name);

  const classPrefix = useMemo(() => props.classPrefix || 'van-icon', [props.classPrefix]);

  return (
    <Badge
      tag={tag}
      className={classnames(
        className,
        classPrefix,
        imageIcon ? '' : `${props.classPrefix}-${name}`,
      )}
      style={{
        color: props.color,
        fontSize: addUnit(props.size),
        ...props.style,
      }}
      onClick={onClick}
      onTouchStart={onTouchStart}
      {...props.badge}
    >
      {props?.children}
      {imageIcon && <img className={classnames('van-icon__image')} src={name} alt={name} />}
    </Badge>
  );
};

Icon.defaultProps = {
  classPrefix: 'van-icon',
  tag: 'i',
};

export default Icon;
