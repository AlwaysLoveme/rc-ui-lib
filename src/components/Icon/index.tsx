import React from 'react';
import { addUnit } from '@/utils';
import classnames from 'classnames';
import { IconProps } from './props';
import Badge from '../badge';
import './style/index.scss';

export const isImage = (name?: string): boolean => {
    return name ? name.indexOf('/') !== -1 : false;
};
const baseClass = 'r-icon';

const Icon: React.FC<IconProps> = (props) => {
    const {
        tag = 'i',
        name,
        badge,
        classPrefix = 'r-icon',
        color,
        size,
        dot = false,
        className,
        onClick,
        children
    } = props;

    const isImageIcon = isImage(name);

    const handleClick = (e: TouchEvent | MouseEvent) => {
        onClick?.(e);
    };

    return (
        <Badge
            className={classnames(
                classPrefix,
                { [`${classPrefix}-${name}`]: !isImageIcon },
                className
            )}
            content={badge}
            dot={dot}
            onClick={handleClick}
            style={{
                color,
                fontSize: addUnit(size)
            }}
            tag={tag}
        >
            {children}
            {isImageIcon && <img className={classnames(`${baseClass}__image`)} src={name} />}
        </Badge>
    );
};

export default Icon;
