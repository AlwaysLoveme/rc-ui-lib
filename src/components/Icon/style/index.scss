@import '../../style/encode-woff2.scss';

.r-icon {
    // position: relative;
    // transform: translateY(20%);
    &__image {
        width: 1em;
        height: 1em;
        object-fit: contain;
    }
}
