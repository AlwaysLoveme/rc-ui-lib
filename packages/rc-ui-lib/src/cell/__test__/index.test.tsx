import React from 'react';
import { Cell } from '../index';
import CellGroup from '../CellGroup';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Icon from '../../icon';


describe("Cell and CellGroup", () => {

    it('should render value correctly', () => {
        const wrapper = mount(
            <Cell value="Custom Value"></Cell>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render title correctly', () => {
        const wrapper = mount(
            <Cell title="Custom Title"></Cell>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render label correctly', () => {
        const wrapper = mount(
            <Cell
                title="Custom Title"
                value="Custom Value"
                label="Custom Label">
            </Cell>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render icon correctly', () => {
        const wrapper = mount(
            <Cell title="单元格" icon="shop-o" />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render extra correctly', () => {
        const wrapper = mount(
            <Cell title="单元格"
                icon="shop-o"
                rightIcon={<Icon name="search" className="search-icon" />}
                extra="extra"
            >
            </Cell>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should change arrow direction when using arrowDirection prop', () => {
        const wrapper = mount(
            <Cell title="title" value="value" isLink arrowDirection="down"></Cell>
        );
        expect(wrapper.find('.rc-cell__right-icon')).toMatchSnapshot();
    });

    it('should change title style when using titleStyle prop', () => {
        const wrapper = mount(
            <Cell title="title" titleStyle={{ color: "red" }} />
        );
        const title = wrapper.find('.rc-cell__title');
        expect(title.props().style.color).toEqual('red');
    });

    it('should change icon class prefix when using iconPrefix prop', () => {
        const wrapper = mount(
            <Cell icon='success' iconPrefix="my-icon" />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should allow to disable clicakble when using isLink prop', () => {
        const wrapper = mount(
            <Cell clickable={false} isLink />
        );
        expect(wrapper.hasClass('rc-cell--clickable')).toBeFalsy();
    });

    it('should be wrappered by CellGroup', () => {
        const wrapper = mount(
            <CellGroup>
                <Cell title="单元格" value="内容" />
                <Cell title="单元格" value="内容" label="描述信息" />
            </CellGroup>

        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });


    it('should be wrappered by CellGroup with prop title', () => {
        const wrapper = mount(
            <CellGroup title="单元组">
                <Cell title="单元格" value="内容" />
                <Cell title="单元格" value="内容" label="描述信息" />
            </CellGroup>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should be wrappered by CellGroup with prop border is false', () => {
        const wrapper = mount(
            <CellGroup title="单元组" border={false}>
                <Cell title="单元格" value="内容" />
                <Cell title="单元格" value="内容" label="描述信息" />
            </CellGroup>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should be wrappered by CellGroup with prop inset', () => {
        const wrapper = mount(
            <CellGroup inset>
                <Cell title="单元格" value="内容" />
                <Cell title="单元格" value="内容" label="描述信息" />
            </CellGroup>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    })

});
