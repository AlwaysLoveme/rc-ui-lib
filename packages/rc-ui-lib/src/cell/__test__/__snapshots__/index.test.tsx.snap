// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell and CellGroup should be wrappered by CellGroup 1`] = `
<CellGroup
  border={true}
>
  <div
    className="rc-cell-group rc-hairline--top-bottom"
  >
    <Cell
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
    <Cell
      label="描述信息"
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
          <div
            className="rc-cell__label"
          >
            描述信息
          </div>
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
  </div>
</CellGroup>
`;

exports[`Cell and CellGroup should be wrappered by CellGroup with prop border is false 1`] = `
<CellGroup
  border={false}
  title="单元组"
>
  <div
    className="rc-cell-group__title"
  >
    单元组
  </div>
  <div
    className="rc-cell-group"
  >
    <Cell
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
    <Cell
      label="描述信息"
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
          <div
            className="rc-cell__label"
          >
            描述信息
          </div>
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
  </div>
</CellGroup>
`;

exports[`Cell and CellGroup should be wrappered by CellGroup with prop inset 1`] = `
<CellGroup
  border={true}
  inset={true}
>
  <div
    className="rc-cell-group rc-cell-group--inset"
  >
    <Cell
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
    <Cell
      label="描述信息"
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
          <div
            className="rc-cell__label"
          >
            描述信息
          </div>
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
  </div>
</CellGroup>
`;

exports[`Cell and CellGroup should be wrappered by CellGroup with prop title 1`] = `
<CellGroup
  border={true}
  title="单元组"
>
  <div
    className="rc-cell-group__title"
  >
    单元组
  </div>
  <div
    className="rc-cell-group rc-hairline--top-bottom"
  >
    <Cell
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
    <Cell
      label="描述信息"
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
          <div
            className="rc-cell__label"
          >
            描述信息
          </div>
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
  </div>
</CellGroup>
`;

exports[`Cell and CellGroup should change arrow direction when using arrowDirection prop 1`] = `
Array [
  <Icon
    className="rc-cell__right-icon"
    classPrefix="van-icon"
    name="arrow-down"
    tag="i"
  >
    <Badge
      className="rc-cell__right-icon van-icon van-icon-arrow-down"
      showZero={true}
      style={
        Object {
          "color": undefined,
          "fontSize": undefined,
        }
      }
      tag="i"
    >
      <i
        className="rc-badge__wrapper rc-cell__right-icon van-icon van-icon-arrow-down"
        style={
          Object {
            "color": undefined,
            "fontSize": undefined,
          }
        }
      />
    </Badge>
  </Icon>,
  <Badge
    className="rc-cell__right-icon van-icon van-icon-arrow-down"
    showZero={true}
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
    tag="i"
  >
    <i
      className="rc-badge__wrapper rc-cell__right-icon van-icon van-icon-arrow-down"
      style={
        Object {
          "color": undefined,
          "fontSize": undefined,
        }
      }
    />
  </Badge>,
  <i
    className="rc-badge__wrapper rc-cell__right-icon van-icon van-icon-arrow-down"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  />,
]
`;

exports[`Cell and CellGroup should change icon class prefix when using iconPrefix prop 1`] = `
<Cell
  icon="success"
  iconPrefix="my-icon"
>
  <div
    className="rc-cell"
  >
    <Icon
      className="rc-cell__left-icon"
      classPrefix="my-icon"
      name="success"
      tag="i"
    >
      <Badge
        className="rc-cell__left-icon my-icon my-icon-success"
        showZero={true}
        style={
          Object {
            "color": undefined,
            "fontSize": undefined,
          }
        }
        tag="i"
      >
        <i
          className="rc-badge__wrapper rc-cell__left-icon my-icon my-icon-success"
          style={
            Object {
              "color": undefined,
              "fontSize": undefined,
            }
          }
        />
      </Badge>
    </Icon>
  </div>
</Cell>
`;

exports[`Cell and CellGroup should render extra correctly 1`] = `
<Cell
  extra="extra"
  icon="shop-o"
  rightIcon={
    <Icon
      className="search-icon"
      classPrefix="van-icon"
      name="search"
      tag="i"
    />
  }
  title="单元格"
>
  <div
    className="rc-cell"
  >
    <Icon
      className="rc-cell__left-icon"
      classPrefix="van-icon"
      name="shop-o"
      tag="i"
    >
      <Badge
        className="rc-cell__left-icon van-icon van-icon-shop-o"
        showZero={true}
        style={
          Object {
            "color": undefined,
            "fontSize": undefined,
          }
        }
        tag="i"
      >
        <i
          className="rc-badge__wrapper rc-cell__left-icon van-icon van-icon-shop-o"
          style={
            Object {
              "color": undefined,
              "fontSize": undefined,
            }
          }
        />
      </Badge>
    </Icon>
    <div
      className="rc-cell__title"
    >
      单元格
    </div>
    <Icon
      className="search-icon"
      classPrefix="van-icon"
      name="search"
      tag="i"
    >
      <Badge
        className="search-icon van-icon van-icon-search"
        showZero={true}
        style={
          Object {
            "color": undefined,
            "fontSize": undefined,
          }
        }
        tag="i"
      >
        <i
          className="rc-badge__wrapper search-icon van-icon van-icon-search"
          style={
            Object {
              "color": undefined,
              "fontSize": undefined,
            }
          }
        />
      </Badge>
    </Icon>
    extra
  </div>
</Cell>
`;

exports[`Cell and CellGroup should render icon correctly 1`] = `
<Cell
  icon="shop-o"
  title="单元格"
>
  <div
    className="rc-cell"
  >
    <Icon
      className="rc-cell__left-icon"
      classPrefix="van-icon"
      name="shop-o"
      tag="i"
    >
      <Badge
        className="rc-cell__left-icon van-icon van-icon-shop-o"
        showZero={true}
        style={
          Object {
            "color": undefined,
            "fontSize": undefined,
          }
        }
        tag="i"
      >
        <i
          className="rc-badge__wrapper rc-cell__left-icon van-icon van-icon-shop-o"
          style={
            Object {
              "color": undefined,
              "fontSize": undefined,
            }
          }
        />
      </Badge>
    </Icon>
    <div
      className="rc-cell__title"
    >
      单元格
    </div>
  </div>
</Cell>
`;

exports[`Cell and CellGroup should render label correctly 1`] = `
<Cell
  label="Custom Label"
  title="Custom Title"
  value="Custom Value"
>
  <div
    className="rc-cell"
  >
    <div
      className="rc-cell__title"
    >
      Custom Title
      <div
        className="rc-cell__label"
      >
        Custom Label
      </div>
    </div>
    <div
      className="rc-cell__value"
    >
      <span>
        Custom Value
      </span>
    </div>
  </div>
</Cell>
`;

exports[`Cell and CellGroup should render title correctly 1`] = `
<Cell
  title="Custom Title"
>
  <div
    className="rc-cell"
  >
    <div
      className="rc-cell__title"
    >
      Custom Title
    </div>
  </div>
</Cell>
`;

exports[`Cell and CellGroup should render value correctly 1`] = `
<Cell
  value="Custom Value"
>
  <div
    className="rc-cell"
  >
    <div
      className="rc-cell__value rc-cell__value--alone"
    >
      <span>
        Custom Value
      </span>
    </div>
  </div>
</Cell>
`;

exports[`should be wrappered by CellGroup 1`] = `
<CellGroup
  border={true}
>
  <div
    className="rc-cell-group rc-hairline--top-bottom"
  >
    <Cell
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
    <Cell
      label="描述信息"
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
          <div
            className="rc-cell__label"
          >
            描述信息
          </div>
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
  </div>
</CellGroup>
`;

exports[`should be wrappered by CellGroup with prop border is false 1`] = `
<CellGroup
  border={false}
  title="单元组"
>
  <div
    className="rc-cell-group__title"
  >
    单元组
  </div>
  <div
    className="rc-cell-group"
  >
    <Cell
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
    <Cell
      label="描述信息"
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
          <div
            className="rc-cell__label"
          >
            描述信息
          </div>
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
  </div>
</CellGroup>
`;

exports[`should be wrappered by CellGroup with prop inset 1`] = `
<CellGroup
  border={true}
  inset={true}
>
  <div
    className="rc-cell-group rc-cell-group--inset"
  >
    <Cell
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
    <Cell
      label="描述信息"
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
          <div
            className="rc-cell__label"
          >
            描述信息
          </div>
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
  </div>
</CellGroup>
`;

exports[`should be wrappered by CellGroup with prop title 1`] = `
<CellGroup
  border={true}
  title="单元组"
>
  <div
    className="rc-cell-group__title"
  >
    单元组
  </div>
  <div
    className="rc-cell-group rc-hairline--top-bottom"
  >
    <Cell
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
    <Cell
      label="描述信息"
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
          <div
            className="rc-cell__label"
          >
            描述信息
          </div>
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
  </div>
</CellGroup>
`;

exports[`should be wrappered by CellGroup with prop title and value 1`] = `
<CellGroup
  border={true}
  title="单元组"
>
  <div
    className="rc-cell-group__title"
  >
    单元组
  </div>
  <div
    className="rc-cell-group rc-hairline--top-bottom"
  >
    <Cell
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
    <Cell
      label="描述信息"
      title="单元格"
      value="内容"
    >
      <div
        className="rc-cell"
      >
        <div
          className="rc-cell__title"
        >
          单元格
          <div
            className="rc-cell__label"
          >
            描述信息
          </div>
        </div>
        <div
          className="rc-cell__value"
        >
          <span>
            内容
          </span>
        </div>
      </div>
    </Cell>
  </div>
</CellGroup>
`;

exports[`should change arrow direction when using arrowDirection prop 1`] = `
Array [
  <Icon
    className="rc-cell__right-icon"
    classPrefix="van-icon"
    name="arrow-down"
    tag="i"
  >
    <Badge
      className="rc-cell__right-icon van-icon van-icon-arrow-down"
      showZero={true}
      style={
        Object {
          "color": undefined,
          "fontSize": undefined,
        }
      }
      tag="i"
    >
      <i
        className="rc-badge__wrapper rc-cell__right-icon van-icon van-icon-arrow-down"
        style={
          Object {
            "color": undefined,
            "fontSize": undefined,
          }
        }
      />
    </Badge>
  </Icon>,
  <Badge
    className="rc-cell__right-icon van-icon van-icon-arrow-down"
    showZero={true}
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
    tag="i"
  >
    <i
      className="rc-badge__wrapper rc-cell__right-icon van-icon van-icon-arrow-down"
      style={
        Object {
          "color": undefined,
          "fontSize": undefined,
        }
      }
    />
  </Badge>,
  <i
    className="rc-badge__wrapper rc-cell__right-icon van-icon van-icon-arrow-down"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  />,
]
`;

exports[`should change icon class prefix when using iconPrefix prop 1`] = `
<Cell
  icon="success"
  iconPrefix="my-icon"
>
  <div
    className="rc-cell"
  >
    <Icon
      className="rc-cell__left-icon"
      classPrefix="my-icon"
      name="success"
      tag="i"
    >
      <Badge
        className="rc-cell__left-icon my-icon my-icon-success"
        showZero={true}
        style={
          Object {
            "color": undefined,
            "fontSize": undefined,
          }
        }
        tag="i"
      >
        <i
          className="rc-badge__wrapper rc-cell__left-icon my-icon my-icon-success"
          style={
            Object {
              "color": undefined,
              "fontSize": undefined,
            }
          }
        />
      </Badge>
    </Icon>
  </div>
</Cell>
`;

exports[`should render extra correctly 1`] = `
<Cell
  extra="extra"
  icon="shop-o"
  rightIcon={
    <Icon
      className="search-icon"
      classPrefix="van-icon"
      name="search"
      tag="i"
    />
  }
  title="单元格"
>
  <div
    className="rc-cell"
  >
    <Icon
      className="rc-cell__left-icon"
      classPrefix="van-icon"
      name="shop-o"
      tag="i"
    >
      <Badge
        className="rc-cell__left-icon van-icon van-icon-shop-o"
        showZero={true}
        style={
          Object {
            "color": undefined,
            "fontSize": undefined,
          }
        }
        tag="i"
      >
        <i
          className="rc-badge__wrapper rc-cell__left-icon van-icon van-icon-shop-o"
          style={
            Object {
              "color": undefined,
              "fontSize": undefined,
            }
          }
        />
      </Badge>
    </Icon>
    <div
      className="rc-cell__title"
    >
      单元格
    </div>
    <Icon
      className="search-icon"
      classPrefix="van-icon"
      name="search"
      tag="i"
    >
      <Badge
        className="search-icon van-icon van-icon-search"
        showZero={true}
        style={
          Object {
            "color": undefined,
            "fontSize": undefined,
          }
        }
        tag="i"
      >
        <i
          className="rc-badge__wrapper search-icon van-icon van-icon-search"
          style={
            Object {
              "color": undefined,
              "fontSize": undefined,
            }
          }
        />
      </Badge>
    </Icon>
    extra
  </div>
</Cell>
`;

exports[`should render icon correctly 1`] = `
<Cell
  icon="shop-o"
  title="单元格"
>
  <div
    className="rc-cell"
  >
    <Icon
      className="rc-cell__left-icon"
      classPrefix="van-icon"
      name="shop-o"
      tag="i"
    >
      <Badge
        className="rc-cell__left-icon van-icon van-icon-shop-o"
        showZero={true}
        style={
          Object {
            "color": undefined,
            "fontSize": undefined,
          }
        }
        tag="i"
      >
        <i
          className="rc-badge__wrapper rc-cell__left-icon van-icon van-icon-shop-o"
          style={
            Object {
              "color": undefined,
              "fontSize": undefined,
            }
          }
        />
      </Badge>
    </Icon>
    <div
      className="rc-cell__title"
    >
      单元格
    </div>
  </div>
</Cell>
`;

exports[`should render label correctly 1`] = `
<Cell
  label="Custom Label"
  title="Custom Title"
  value="Custom Value"
>
  <div
    className="rc-cell"
  >
    <div
      className="rc-cell__title"
    >
      Custom Title
      <div
        className="rc-cell__label"
      >
        Custom Label
      </div>
    </div>
    <div
      className="rc-cell__value"
    >
      <span>
        Custom Value
      </span>
    </div>
  </div>
</Cell>
`;

exports[`should render title correctly 1`] = `
<Cell
  title="Custom Title"
>
  <div
    className="rc-cell"
  >
    <div
      className="rc-cell__title"
    >
      Custom Title
    </div>
  </div>
</Cell>
`;

exports[`should render value correctly 1`] = `
<Cell
  value="Custom Value"
>
  <div
    className="rc-cell"
  >
    <div
      className="rc-cell__value rc-cell__value--alone"
    >
      <span>
        Custom Value
      </span>
    </div>
  </div>
</Cell>
`;
